lib/netext/dialer_test.go

```diff
@@ -0,0 +1,52 @@
+package netext
+
+import (
+	"net"
+	"testing"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/stretchr/testify/assert"
+)
+
+type testResolver struct {
+	hosts map[string]net.IP
+}
+
+func (r testResolver) FetchOne(host string) (net.IP, error) { return r.hosts[host], nil }
+
+func TestDialerAddrWithDNSResolver(t *testing.T) {
+	dialer := newDialerWithResolver(net.Dialer{}, newResolver())
+
+	addr, err := dialer.dialAddr("example.com:80")
+	assert.NoError(t, err)
+	assert.Equal(t, "1.2.3.4:80", addr)
+}
+
+func TestDialerAddrWithCachedHost(t *testing.T) {
+	dialer := newDialerWithResolver(net.Dialer{}, newResolver())
+	dialer.Hosts = map[string]lib.IPPort{
+		"example.com":      {IP: net.ParseIP("3.4.5.6"), Port: ""},
+		"example.com:443":  {IP: net.ParseIP("3.4.5.6"), Port: "8443"},
+		"example.com:8080": {IP: net.ParseIP("3.4.5.6"), Port: "9090"},
+	}
+
```

Comment:
can you combine the two tests, by adding another domain without a "binding" without a port so it gets resolved by the testResolver instead of the `hosts` binding? 

Also maybe have a test of the `blacklist`  in combination with `hosts` ? Maybe just a third domain with default resolving to a blacklisted IP, a `hosts` binding that is blacklisted again and one that isn't?